File: frontend/src/types/project.ts
import type { GeneratedContent } from './generation';

export interface Project {
  id: string;
  user_id: string;
  title: string;
  description: string;
  target_audience: string;
  desired_tone: string;
  original_video_filename: string;
  video_path: string;
  transcript: string;
  analysis: Record<string, unknown>;
  generated_content: GeneratedContent;
  selected_template: string;
  ai_provider_used: string;
  generation_error: string;
  status: ProjectStatus;
  created_at: string;
  updated_at: string;
}

export type ProjectStatus = 'draft' | 'uploaded' | 'transcribing' | 'analyzing' | 'generated' | 'failed';

export interface CreateProjectData {
  title: string;
  description: string;
  target_audience: string;
  desired_tone: string;
}

File: backend/app/schemas/project.py
from typing import Any

from pydantic import BaseModel


class CreateProjectRequest(BaseModel):
    title: str
    description: str = ""
    target_audience: str = ""
    desired_tone: str = ""


class UpdateProjectRequest(BaseModel):
    title: str | None = None
    description: str | None = None
    target_audience: str | None = None
    desired_tone: str | None = None
    selected_template: str | None = None


class UpdateContentRequest(BaseModel):
    generated_content: dict[str, Any]


class ProjectResponse(BaseModel):
    id: str
    user_id: str
    title: str
    description: str
    target_audience: str
    desired_tone: str
    original_video_filename: str
    video_path: str
    transcript: str
    analysis: dict[str, Any]
    generated_content: dict[str, Any]
    selected_template: str
    ai_provider_used: str
    generation_error: str
    status: str
    created_at: str
    updated_at: str
